frontend/app/routes/protected/renew/file-taxes.tsx

```diff
@@ -0,0 +1,99 @@
+import type { FormEvent } from 'react';
+
+import type { ActionFunctionArgs, LoaderFunctionArgs, MetaFunction } from '@remix-run/node';
+import { redirect } from '@remix-run/node';
+import { useFetcher, useLoaderData } from '@remix-run/react';
+
+import { Trans, useTranslation } from 'react-i18next';
+
+import { TYPES } from '~/.server/constants';
+import { ButtonLink } from '~/components/buttons';
+import { InlineLink } from '~/components/inline-link';
+import { LoadingButton } from '~/components/loading-button';
+import { pageIds } from '~/page-ids';
+import { clearProtectedRenewState, loadProtectedRenewState } from '~/route-helpers/protected-renew-route-helpers.server';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+import { getFixedT } from '~/utils/locale-utils.server';
+import { getLogger } from '~/utils/logging.server';
+import { mergeMeta } from '~/utils/meta-utils';
+import type { RouteHandleData } from '~/utils/route-utils';
+import { getTitleMetaTags } from '~/utils/seo-utils';
+
+export const handle = {
+  i18nNamespaces: getTypedI18nNamespaces('protected-renew', 'gcweb'),
+  pageIdentifier: pageIds.public.renew.fileYourTaxes,
+  pageTitleI18nKey: 'protected-renew:file-your-taxes.page-title',
+} as const satisfies RouteHandleData;
+
+export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+
+export async function loader({ context: { appContainer, session }, params, request }: LoaderFunctionArgs) {
+  const { id } = loadProtectedRenewState({ params, session });
```

Comment:
FYI every `loader` and `action` in the `protected` area will need to verify that the user has been authenticated via RAOIDC by adding:

```typescript
  const raoidcService = await getRaoidcService();
  await raoidcService.handleSessionValidation(request, session);
```

Reply:
Might be a nit pick, but for this particular route/loader, the loading of the state should happen after the authentication check to avoid unnecessary processing for unauthorized users.